### DISCUSSION TITLE
Stop Rich auto-colouring integers in default prints

### DISCUSSION THREAD

**Discussion by bee-san** [Textualize/rich#957] [Q&A] [answered]
Hey!
![image](https://user-images.githubusercontent.com/10378052/105618185-698d7a80-5ddc-11eb-8a30-804af870b2bd.png)

Rich auto-colours the integer `16`, how do I prevent this? I know exactly what colours + text I want to colour, so auto-colouring harms the experience :( ✨

I'm on 3.8 with the latest version of Rich from PyPi :)

**Comment by willmcgugan** [comment, marked as answer]
See https://rich.readthedocs.io/en/latest/highlighting.html for how to disable highlighting.

**Comment by wiz-rd** [comment]
@willmcgugan Is it possible to simply pass through the `highlight` argument to the `Console.print` directly below it? I was looking at the source code and adding this took only a few seconds.

For example:

**Before:**

```python
def print(
    *objects: Any,
    sep: str = " ",
    end: str = "\n",
    file: Optional[IO[str]] = None,
    flush: bool = False,
) -> None:
    r"""Print object(s) supplied via positional arguments.
    This function has an identical signature to the built-in print.
    For more advanced features, see the :class:`~rich.console.Console` class.

    Args:
        sep (str, optional): Separator between printed objects. Defaults to " ".
        end (str, optional): Character to write at end of output. Defaults to "\\n".
        file (IO[str], optional): File to write to, or None for stdout. Defaults to None.
        flush (bool, optional): Has no effect as Rich always flushes output. Defaults to False.

    """
    from .console import Console

    write_console = get_console() if file is None else Console(file=file)
    return write_console.print(*objects, sep=sep, end=end)
```

**After:**

```python
def print(
    *objects: Any,
    sep: str = " ",
    end: str = "\n",
    file: Optional[IO[str]] = None,
    flush: bool = False,
    highlight: bool = True,
) -> None:
    r"""Print object(s) supplied via positional arguments.
    This function has a near-identical signature to the built-in print.
    For more advanced features, see the :class:`~rich.console.Console` class.

    Args:
        sep (str, optional): Separator between printed objects. Defaults to " ".
        end (str, optional): Character to write at end of output. Defaults to "\\n".
        file (IO[str], optional): File to write to, or None for stdout. Defaults to None.
        flush (bool, optional): Has no effect as Rich always flushes output. Defaults to False.
        highlight (bool, optional): Highlight special characters and numbers.

    """
    from .console import Console

    write_console = get_console() if file is None else Console(file=file)
    return write_console.print(*objects, sep=sep, end=end, highlight=highlight)
```

Note that there are only around 30 characters changed, if that.

Is there a reason this cannot be implemented? I believe it would make things a lot easier for the average user and not require them to use the `Console.print(highlight=False)` workaround you suggested.

My apologies if posting on an old discussion like this is unwarranted. I initially intended to just make a pull request solving it myself but I cannot do that without permission, which I cannot do without an issue, which I cannot do without a discussion. And I cannot make a new discussion without first ensuring there isn't an existing one, so that's how I ended up here.

I understand this would not match Python's default `print` signature exactly but it would still be a drop-in replacement and even if that's undesirable, using `kwargs` would, again, still allow `rich.print` to be a drop-in replacement while adding this functionality. I don't think matching the original stdlib `print` is a hill worth dying on at the expense of all else, but since I'm not familiar with Python's core functions signatures may be more important than I realize. So if I am astray in any regard, please let me know how.

I, personally, favor convenience and progress over tradition for tradition's sake, so if we're following Python's signature just because, I would like to ask why we can't change that. But if there's some underlying reason, then, of course, I respect that.

As an alternative, disabling default highlighting outright may be the best of both worlds - `return write_console.print(*objects, sep=sep, end=end, highlight=False)` - as this would prevent this (occasionally) seemingly-unwanted, unfixable frustration (see [this SO discussion](https://stackoverflow.com/questions/66439196/when-i-use-the-rich-library-to-add-colour-why-does-it-affect-the-symbols-around) you replied to) without needing to change `rich.print`'s signature.

Edit: P.S., `**kwargs` and `kwargs.pop("highlight", None)` may also be an option.

Thank you!
Wiz-rd
